best-practices/native/react-native/common-components/buttons/button.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { StyleSheet, TouchableOpacity, Text, Image } from "react-native";
+import PropTypes from "prop-types";
+
+const styles = StyleSheet.create({
+  button: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    borderRadius: 4,
+    marginTop: 5,
+    marginBottom: 5,
+  },
+  text: {
+    color: "#FFF",
+    fontFamily: "oswald",
+  },
+  icon: {
+    position: "absolute",
+    right: "85%",
+  },
+});
```

Comment:
If we needed to add/edit the styles of the button, how would this work in this sort of setup? 

Just wondering if `styles` could be a Dependency Injected component.

Reply:
We could pass in a prop object for styles and set this to 
```
const styles = Stylesheet.create(props.styles || {...})
```

Reply:
I like it!

Reply:
This has been implemented but a little different than I said here. LMK what you think.

Reply:
`defaultStyles` looks good and explicit.